Cargo.toml

```diff
@@ -29,8 +29,8 @@ blake2b_simd = "1"
 ff = "0.13"
 fpe = "0.6"
 group = { version = "0.13", features = ["wnaf-memuse"] }
-halo2_gadgets = "0.3"
-halo2_proofs = { version = "0.3", default-features = false, features = ["batch", "floor-planner-v1-legacy-pdqsort"] }
+halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "improve-backward-compatability-without-zsa" }
```

Comment:
In general, we do not merge PRs that change dependencies in this way, for two reasons:
- It is a security issue to depend on a branch rather than a commit.
- We use `[patch.crates-io]` instead for patching in-progress changes (and we then propagate those sections through the various downstream repos).

Reply:
I updated Cargo.toml to use patch.crates-io for halo2 repos.